ed_sensor_integration/include/ed_sensor_integration/tools/snapshot.h

```
@@ -0,0 +1,189 @@
+namespace ed_sensor_integration {
+
+struct Snapshot
+{
+    rgbd::ImagePtr image;
+    geo::Pose3D sensor_pose;
+};
+
+bool readImage(const std::string& filename, rgbd::ImagePtr& image, geo::Pose3D& sensor_pose)
+{
+    tue::config::DataPointer meta_data;
+
+    try
+    {
+        meta_data = tue::config::fromFile(filename);
+    }
+    catch (tue::config::ParseException& e)
+    {
+        std::cerr << "Could not open '" << filename << "'.\n\n" << e.what() << std::endl;
+        return false;
+    }
+
+    tue::config::Reader r(meta_data);
+
+    // Read image
+    std::string rgbd_filename;
+    if (r.value("rgbd_filename", rgbd_filename))
+    {
+        tue::filesystem::Path abs_rgbd_filename = tue::filesystem::Path(filename).parentPath().join(rgbd_filename);
+
+        std::ifstream f_rgbd;
+        f_rgbd.open(abs_rgbd_filename.string().c_str(), std::ifstream::binary);
+
+        if (!f_rgbd.is_open())
+        {
+            std::cerr << "Could not open '" << filename << "'." << std::endl;
+            return false;
+        }
+
+        image.reset(new rgbd::Image);
+
+        tue::serialization::InputArchive a_in(f_rgbd);
+        rgbd::deserialize(a_in, *image);
+    }
+
+    // Read sensor pose
+    if (!ed::deserialize(r, "sensor_pose", sensor_pose))
+    {
+        std::cerr << "No field 'sensor_pose' specified." << std::endl;
+        return false;
+    }
+
+    return true;
+}
+
+bool loadWorldModel(const std::string& model_name, ed::WorldModel& world_model)
+{
+    ed::UpdateRequest req;
+
+    ed::models::ModelLoader model_loader;
+
+    std::stringstream error;
+    if (!model_loader.create("_root", model_name, req, error, true))
+    {
+        std::cerr << "Model '" << model_name << "' could not be loaded:" << std::endl << std::endl;
+        std::cerr << error.str() << std::endl;
+        return false;
+    }
+
+    // Reset world
+    world_model = ed::WorldModel();
+
+    // Update world
+    world_model.update(req);
+
+    return true;
+}
```

I think returning a WorldModel pointer would be better. Succes would be indicated by it being a nullptr or not.

I am not even sure it is allowed to overwrite a reference with a new instance. As a reference is a pointer to an instance. But I am not 100% sure, need to check cpp docs.

---

Or let it throw an exception. If someone forgets to check the boolean return value the program might give errors later on, making it hard to debug.